pkg/providers/gcp/gcp.go

```diff
@@ -144,12 +198,12 @@ func (g *GCP) GetMetricsForInstances(
 		// Load the cache
 		// TODO make this more explicit, im not sure why this
 		// is needed as we dont use the cache anywhere
-		cachedInstance := g.cache.Get(zone, gceService, instanceName)
-		if cachedInstance == nil {
+		cachedInstance, ok := g.cache.Get(cacheKey(zone, service, instanceName))
+		if cachedInstance == nil && !ok {
```

Comment:
Should it be `||` ? What if a nil value get's cached (if that's even possible).

Reply:
honestly I think this needs to change regardles, once we cahce the instance then i will update this. but for this instacne, if ok is not true then CachedInstance should be null so && works